### PR TITLE
Add Molecule testing

### PR DESCRIPTION
**Pull request by russoz**
Hi @gekmihesg 

I have been using your role constantly in my home setup, thanks! As a matter of fact I am planning on releasing a role that will require this one as dependency. As such I have been looking a lot into testing, especially into `molecule`. 

So, I thought I could help this role by adding molecule testing on it. This is a work in progress, I just managed to make it run end-to-end today but I have not yet added any meaningful tests. However, just by running `yamllint` I was able to pick on some improvements. I am ready to add `ansible-lint` as well, but on a trial run a bunch of nit-picking popped up, so I decided to defer that to another PR.

To run the tests, it requires you to have `dockerd` running and accessible, then in your venv install `molecule[docker]` and `yamllint`. I *think* that will do the trick, then go to the top level directory and run:

    molecule test      # for the complete run
    molecule lint      # to run only yamllint only
    molecule converge  # to run it multiple times without destroying the running container

TO-DO:
* ~~add some real tests~~
* add github worklfows to test it
* ~~test against multiple versions of the docker image~~

Hope you like it :-) 

Cheers
Alex

### CODE REVIEW

**Comment by russoz** [comment]
Hi @gekmihesg just a friendly ping, in case you missed this PR.

File: molecule/default/converge.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+- name: Converge
+  hosts: all
+  gather_facts: false
+  tasks:
```
Comment (gekmihesg) [inline molecule/default/converge.yml:5]:
```suggestion
  tasks:
    - name: "Create /var/lock"
      ansible.builtin.raw: "mkdir -p /var/lock"
      changed_when: false
```

File: molecule/default/molecule.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+driver:
+  name: docker
+platforms:
+  - name: instance_latest
+    image: openwrtorg/rootfs:latest
+    pre_build_image: false
+    groups:
+      - openwrt
+    environment:
+      openwrt_image_tag: latest
```
Comment (gekmihesg) [inline molecule/default/molecule.yml:14]:
```suggestion
    command: sleep infinity
    pre_build_image: true
    groups:
      - openwrt
```

File: molecule/default/molecule.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  - name: instance_x86_64-22.03.2
+    image: openwrtorg/rootfs:x86_64-22.03.2
+    pre_build_image: false
+    groups:
+      - openwrt
+    environment:
+      openwrt_image_tag: x86_64-22.03.2
```
Comment (gekmihesg) [inline molecule/default/molecule.yml:21]:
```suggestion
    command: sleep infinity
    pre_build_image: true
    groups:
      - openwrt
```

File: molecule/default/molecule.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  - name: instance_x86_64-21.02.5
+    image: openwrtorg/rootfs:x86_64-21.02.5
+    pre_build_image: false
+    groups:
+      - openwrt
+    environment:
+      openwrt_image_tag: x86_64-21.02.5
```
Comment (gekmihesg) [inline molecule/default/molecule.yml:28]:
```suggestion
    command: sleep infinity
    pre_build_image: true
    groups:
      - openwrt
```

File: molecule/default/molecule.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  - name: instance_x86_64-19.07.10
+    image: openwrtorg/rootfs:x86_64-19.07.10
+    pre_build_image: false
+    groups:
+      - openwrt
+    environment:
+      openwrt_image_tag: x86_64-19.07.10
```
Comment (gekmihesg) [inline molecule/default/molecule.yml:35]:
```suggestion
    command: "sh -c 'while true; do sleep 10000; done'"
    pre_build_image: true
    groups:
      - openwrt
```

File: molecule/default/molecule.yml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  - name: instance_x86-64-18.06.6
+    image: openwrtorg/rootfs:x86-64-18.06.6
+    pre_build_image: false
+    groups:
+      - openwrt
+    environment:
+      openwrt_image_tag: x86-64-18.06.6
```
Comment (gekmihesg) [inline molecule/default/molecule.yml:42]:
```suggestion
    command: "sh -c 'while true; do sleep 10000; done'"
    pre_build_image: true
    groups:
      - openwrt
```

File: molecule/default/Dockerfile.j2 [hunk 7]
```diff
@@ -0,0 +1,5 @@
+FROM openwrtorg/rootfs:{{item.environment.openwrt_image_tag}}
+
+RUN mkdir -p /var/lock \
+  && opkg update \
+  && opkg install bash
```
Comment (gekmihesg) [inline molecule/default/Dockerfile.j2:5]:
```suggestion
```
(delete file)

**Review by gekmihesg — COMMENTED** [review]
Thanks, I like this very much indeed, especially since I haven't really been using OpenWRT for quite some time now and have no test setup.
I've added some minor suggestions that allow using the upstream images directly which speeds up things quite a bit for me.

**Comment by russoz** [comment]
Hi @gekmihesg thanks for getting back on this, and thanks for the contributions, they make the test code simpler indeed.

I've made a couple of changes in the README, but I reckon there's still a lot of room for improvement. And of course, at some point we should have proper tests for the modules themselves.

**Comment by gekmihesg** [comment]
Sure, but this really is a great starting point for implementing more tests. Thank you!

**Pull request merged**